telemetry/src/lib.rs

```diff
@@ -46,20 +81,112 @@ impl TelemetryConfig {
         }
     }
 
-    pub fn try_init(self) -> Result<(), tracing_subscriber::util::TryInitError> {
+    /// Enable OpenTelemetry/OTLP tracing.
+    /// You can optionally provide a service name, version, and environment.
+    /// If omitted, these will default to environment variables (`SERVICE_NAME`, `SERVICE_VERSION`, `ENVIRONMENT`) or hard-coded strings.
+    #[must_use]
+    pub fn with_opentelemetry(
+        mut self,
+        service_name: Option<String>,
```

Comment:
lets make providing a service name along with version and environment mandatory. I cant imagine not wanting that info.